### PR TITLE
lustre supported

### PR DESCRIPTION
**Pull request by YinYangOfDao**
the code is backward compatible, but pay attention that you might want to avoid generating duplicate new-schema mounting items when old-schema already mounted. may need to either 1. umount old version first or 2. remove the duplicate entries in generated new-schema mounting.yaml
TODO:
multiple mdt node support

### CODE REVIEW

File: src/ClusterBootstrap/template/storage/auto_share/auto_share_nfs_v2.py [hunk 1]
```diff
@@ -16,77 +16,69 @@
 import getpass
 import copy
 
-def istrue( config, arg, default=False):
```
Comment (Anbang-Hu) [inline src/ClusterBootstrap/template/storage/auto_share/auto_share_nfs_v2.py:19]: This is no longer used?

Comment (YinYangOfDao) [reply src/ClusterBootstrap/template/storage/auto_share/auto_share_nfs_v2.py:19]: no, no longer used

File: src/ClusterBootstrap/template/cloud-config/cloud_init_lustre.txt.template [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+ - echo "export KUBE_API_SERVER='{{cnf["escaped_api_servers"]}}'" >> boot.env
+ - echo "export MOD_2_CP='{{cnf["file_modules_2_copy"] | join(';')}}'" >> boot.env
+ - echo "export MGS_NODE_PRIVATE_IP='{{cnf["mgs_node_private_ip"]}}'" >> boot.env
+{% if "mdt_id" in cnf %}
+ - echo "export MDT_ID={{cnf["mdt_id"]}}" >> boot.env
+{% endif %}
+{% if "oss_id" in cnf %}
+ - echo "export OSS_ID={{cnf["oss_id"]}}" >> boot.env
+{% endif %}
+ # pull docker and install NFS service kits
```
Comment (Anbang-Hu) [inline src/ClusterBootstrap/template/cloud-config/cloud_init_lustre.txt.template:25]: Lustre

File: src/ClusterBootstrap/template/storage/auto_share/auto_share_nfs_v2.py [hunk 3]
```diff
@@ -111,73 +103,117 @@ def mount_fileshare(verbose=True):
...
+def link_fileshare(verbose=True):
     with open("mounting.yaml", 'r') as datafile:
         config = yaml.load(datafile)
         datafile.close()
-    (retcode, output, err) = exec_with_output("sudo mount")
+    (retcode, output, err) = exec_with_output("sudo mount", verbose=False)
     for mnt_itm in config.values():
+        # to make the code backward compatible
+        client_mount_key = "client_mount_root" if "fileshare_system" in mnt_itm else "remote_mount_path"
+        local_path_key = "storage_local_path" if "fileshare_system" in mnt_itm else "nfs_local_path"
```
Comment (Anbang-Hu) [inline src/ClusterBootstrap/template/storage/auto_share/auto_share_nfs_v2.py:153]: `server_path`

File: src/ClusterBootstrap/template/storage/auto_share/setup_lustre.sh [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+baseurl=https://downloads.hpdd.intel.com/public/lustre/latest-feature-release/el7/client/
+gpgcheck=0
+EOM
+
+
+
+
+yum install -y e2fsprogs
+mkdir /tmp/lustre
+wget -O /tmp/lustre/kernel-3.10.0-1062.1.1.el7_lustre.x86_64.rpm https://downloads.whamcloud.com/public/lustre/lustre-2.13.0/el7.7.1908/server/RPMS/x86_64/kernel-3.10.0-1062.1.1.el7_lustre.x86_64.rpm
```
Comment (Anbang-Hu) [inline src/ClusterBootstrap/template/storage/auto_share/setup_lustre.sh:21]: Can you add detailed deployment instruction on which version of Azure VM to use for Lustre in README?

File: src/ClusterBootstrap/cloud_init_deploy.py [hunk 5]
```diff
@@ -426,55 +436,45 @@ def gen_mounting_yaml(config):
...
-                    lnk_root = v['remote_link_path_root']
+                mnt_root = config["physical-mount-path-vc"] if "vc" in v else config["physical-mount-path"]
+            assert mnt_root not in mount_sources_set, "Duplicate mounting mount path detected:\n{}".format(mnt_root)
+            mount_sources_set.add(mnt_root)
+            if "client_link_root" in v:
+                lnk_root = os.path.join(v["client_link_root"], vc)
+            else:
+                lnk_root = config["dltsdata-storage-mount-path"] if "vc" in v else config["storage-mount-path"]
+            # process leaves
+            if not "client_links" in v or len(v["client_links"]) == 0:
```
Comment (Anbang-Hu) [inline src/ClusterBootstrap/cloud_init_deploy.py:461]: `not in`

**Pull request merged**